packages/cli/src/databases/repositories/sharedWorkflow.repository.ts

```diff
@@ -1,10 +1,31 @@
 import { Service } from 'typedi';
-import { DataSource, Repository } from 'typeorm';
+import { DataSource, type FindOptionsWhere, Repository, In } from 'typeorm';
 import { SharedWorkflow } from '../entities/SharedWorkflow';
+import { type User } from '../entities/User';
 
 @Service()
 export class SharedWorkflowRepository extends Repository<SharedWorkflow> {
 	constructor(dataSource: DataSource) {
 		super(SharedWorkflow, dataSource.manager);
 	}
+
+	async hasAccess(workflowId: string, user: User) {
+		const where: FindOptionsWhere<SharedWorkflow> = {
+			workflowId,
+		};
+		if (!user.hasGlobalScope('workflow:read')) {
+			where.userId = user.id;
+		}
+		return this.exist({ where });
+	}
```

Comment:
I'd find it more intuitive if `hasAccess` were defined on the `User` entity, just like `user.hasGlobalScope()`.

Reply:
that would be nice, but doing so in the current code could create a circular dependency between entities and repositories. I'll think about it, and try to find alternative ways of doing this that feel more intuitive.